transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -1111,9 +1206,34 @@ private WriteAndFlushTask(Recycler.Handle<WriteAndFlushTask> handle) {
         }
 
         @Override
-        public void write(AbstractChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
-            super.write(ctx, msg, promise);
+        public void write(AbstractChannelHandlerContext ctx, Object msg, ChannelPromise promise, boolean autoFlush) {
+            super.write(ctx, msg, promise, autoFlush);
             ctx.invokeFlush();
         }
     }
+
+    // The following code is needed to support auto flush.
+    @Override
+    public void run() {
+        if (WRITE_PENDING_UPDATER.compareAndSet(this, 1, 0)) {
+            // As we first update the write pending flag we may schedule some unnessary flushes in worst-case.
+            // This is ok as will not harm.
+            if (channel().isActive()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1066)